src/systems/camera-system.js

```diff
@@ -0,0 +1,182 @@
+import { waitForDOMContentLoaded } from "../utils/async-utils";
+import { setMatrixWorld } from "../utils/three-utils";
+import { paths } from "./userinput/paths";
+import { getBox } from "../utils/auto-box-collider";
+import qsTruthy from "../utils/qs_truthy";
+
+const calculateViewingDistance = (function() {
+  const center = new THREE.Vector3();
+  return function calculateViewingDistance(camera, object) {
+    const box = getBox(object.el, object.el.getObject3D("mesh"), true);
+    box.getCenter(center);
+    const halfYExtents = Math.max(Math.abs(box.max.y - center.y), Math.abs(center.y - box.min.y));
+    const halfXExtents = Math.max(Math.abs(box.max.x - center.x), Math.abs(center.x - box.min.x));
+    const halfVertFOV = THREE.Math.degToRad(camera.el.sceneEl.camera.fov / 2);
+    const halfHorFOV =
+      Math.atan(Math.tan(halfVertFOV) * camera.el.sceneEl.camera.aspect) * (object.el.sceneEl.is("vr-mode") ? 0.5 : 1);
+    const margin = 1.05;
+    const l1 = Math.abs((halfYExtents * margin) / Math.tan(halfVertFOV));
+    const l2 = Math.abs((halfXExtents * margin) / Math.tan(halfHorFOV));
+    const l3 = Math.abs(box.max.z - center.z) + Math.max(l1, l2);
+    const l = object.el.sceneEl.is("vr-mode") ? Math.max(0.25, l3) : l3;
+    return l;
+  };
+})();
+
+const moveRigSoCameraLooksAtObject = (function() {
+  const cqInv = new THREE.Quaternion();
+  const owq = new THREE.Quaternion();
+  const v1 = new THREE.Vector3();
+  const v2 = new THREE.Vector3();
+  const owp = new THREE.Vector3();
+
+  return function moveRigSoCameraLooksAtObject(rig, camera, object) {
+    object.getWorldQuaternion(owq);
+    object.getWorldPosition(owp);
+    cqInv.copy(camera.quaternion).inverse();
+    rig.quaternion.copy(owq).multiply(cqInv);
+
+    v1.set(0, 0, 1)
+      .multiplyScalar(calculateViewingDistance(camera, object))
+      .applyQuaternion(owq);
+    v2.copy(camera.position).applyQuaternion(rig.quaternion);
+    rig.position.subVectors(v1, v2).add(owp);
+
+    rig.matrixNeedsUpdate = true;
+  };
+})();
+
+let numCameraModes = 0;
```

Comment:
this is a bit weird to initialize constants this way -- I'd just write them

Reply:
oic, you need `numCameraModes` -- not a huge deal seems fine all things considered. if it were me i'd probably declare the constants as numbers and make an explicit array that lists them in the order we want them cycled.